app/controllers/straws_controller.rb

```diff
@@ -2,16 +2,59 @@ class StrawsController < ApplicationController
   skip_before_action :verify_authenticity_token, only: :create
 
   def create
+# basic parsing
     straw_usernames = params[:text].split(' ')
-    short_straw_username = straw_usernames.sample
     host_username = params[:user_name]
 
-    straw_response = "@#{host_username} "
-    straw_response += "created a straw poll for #{straw_usernames.to_sentence}.\n\n"
-    straw_response += "*#{short_straw_username} has drawn the short straw*. Sucka!"
+# advanced parsing
+    straw_task_description = params[:text].scan(/\"(.+)\"/).flatten.join
+    straw_selected_users = params[:text].scan(/@[a-zA-Z0-9_\-\.]{1,21}\b/)
+
+# selection logic
+    short_straw_username = straw_selected_users.sample
+    straw_list = ""
+    straw_selected_users.each { |username| straw_list += "#{username}\n" }
```

Comment:
Yeah that totally makes sense. I spent little to no time refactoring, so this got skipped. I think most of this could use more refactoring as well - was thinking its likely to change again though.

Not sure if we should wait on doing a refactoring/reorganizing or not?